modules/api_loader.lua

```diff
@@ -1,81 +1,146 @@
-    local struct = { }
-    for _, item in ipairs(s:children("param")) do
-      struct[item:attr("name")] = item:attributes()
+
+
+ local function LoadEnums(api, dest)
+   for first, v in pairs (dest.interface) do
+    for _, s in ipairs(v.body:children("enum")) do
+      local name = s:attr("name")
+      dest.interface[first].enum[name]={}
+      local i = 1
+      for _,e in ipairs(s:children("element")) do
+        local enum_value = e:attr("name")
+        dest.interface[first].enum[name][enum_value]=i
+        i= i + 1
+      end
     end
-    dest.struct[get_name(s)] = struct
   end
+ end
+ 
+ local function LoadStructs(api, dest)
+   for first, v in pairs (dest.interface) do
+    for _, s in ipairs(v.body:children("struct")) do
+      local name = s:attr("name")
+      local temp_param = {}
+      local temp_func = {}
+      temp_func["name"] = name
+      for _, item in ipairs(s:children("param")) do
+        param_name, param_data = LoadParamsInFunction(item, first)
+        temp_param[param_name] = param_data
+      end
+      temp_func["param"] = temp_param
+      dest.interface[first].struct[name]=temp_func
+    end
+   end
+ end
+ 
 
-  for n, s in pairs(dest.struct) do
-    for _, p in pairs(s) do
-      if type(p.type) == 'string' then
-        if p.type == "Integer" then
-          p.class = dest.classes.Integer
-        elseif p.type == "String" then
-          p.class = dest.classes.String
-        elseif p.type == "Float" then
-          p.class = dest.classes.Float
-        elseif p.type == "Boolean" then
-          p.class = dest.classes.Boolean
-        elseif dest.enum[p.type] then
-          p.class = dest.classes.Enum
-          p.type = dest.enum[p.type]
-        elseif dest.struct[p.type] then
-          p.class = dest.classes.Struct
-          p.type = dest.struct[p.type]
-        end
+local function LoadFunction( api, dest  )
+  for first, v in pairs (dest.interface) do
+    for _, s in ipairs(v.body:children("function")) do
+      local name = s:attr("name")
+      local msg_type = s:attr("messagetype")
+      local temp_func = {}
+      local temp_param = {}
+      temp_func["name"] = name
+      temp_func["messagetype"] = msg_type
+      for _, item in ipairs(s:children("param")) do
+        param_name, param_data = LoadParamsInFunction(item, first)
+        temp_param[param_name] = param_data
       end
+
+      temp_func["param"] = temp_param
+      dest.interface[first].type[msg_type].functions[name]=temp_func
     end
   end
 end
 
-function module.init(path, include_parent_name)
+-- Load interfaces from api. Each function, enum and struct will be 
+-- kept inside appropriate interface
+local function LoadInterfaces( api, dest )
+  local interfaces = api:xpath("//interface")
+  for _, s in ipairs(interfaces) do
+    name = s:attr("name")
+    dest.interface[name] ={}
+    dest.interface[name].body = s
+    dest.interface[name].type={}
+    dest.interface[name].type['request']={}
```

Comment:
@VVeremjova  It is possible to use something like a reference on `dest.interface[name]` and `dest.interface[name].type`?